packages/block-library/src/embed/embed-preview.native.js

```diff
@@ -67,19 +69,51 @@ const EmbedPreview = ( {
 		}
 	}
 
-	// Currently returning a Text component that act's as the Embed Preview to simulate the caption's isSelected state.
+	const parsedHost = new URL( url ).host.split( '.' );
+	const parsedHostBaseUrl = parsedHost
+		.splice( parsedHost.length - 2, parsedHost.length - 1 )
+		.join( '.' );
+
+	const cannotShowThumbnail =
+		! previewable ||
+		! preview ||
+		! preview.thumbnail_url?.length ||
+		! preview.height ||
+		! preview.width;
+
 	return (
 		<TouchableWithoutFeedback
 			accessible={ ! isSelected }
 			onPress={ onEmbedPreviewPress }
 			disabled={ ! isSelected }
 		>
 			<View>
-				<View style={ containerStyle }>
-					<View style={ styles.embed__icon }>{ icon }</View>
-					<Text style={ labelStyle }>{ label }</Text>
-					<Text style={ placeholderTextStyle }>{ preview }</Text>
-				</View>
+				{ cannotShowThumbnail ? (
+					<View style={ containerStyle }>
+						<View style={ styles.embed__icon }>{ icon }</View>
+						<Text style={ labelStyle }>{ label }</Text>
+						<Text style={ placeholderTextStyle }>
+							{ sprintf(
+								/* translators: %s: host providing embed content e.g: www.youtube.com */
+								__(
+									"Embedded content from %s can't be viewed in the mobile editor at the moment. Please preview the page to see the embedded content."
+								),
+								parsedHostBaseUrl
+							) }
+						</Text>
+					</View>
+				) : (
+					<Image
+						style={ {
+							flex: 1,
```

Comment:
What do you think about moving this to a style object?

Reply:
The `aspectRatio` is already dynamically generated so we'll need to pass that in from code again and only `flex: 1` will be in a style object, so maybe we can skip it for now?